Packages/com.unity.inputsystem/InputSystem/InputManager.cs

```diff
@@ -3269,7 +3269,7 @@ internal void RestoreDevicesAfterDomainReload()
                         // to create a device with the same layout.
                         if (!deviceState.description.empty)
                         {
-                            device = AddDevice(deviceState.description, throwIfNoLayoutFound: true,
+                            device = AddDevice(deviceState.description, throwIfNoLayoutFound: false,
```

Comment:
On the fence about this.

Overall, I think we shouldn't silently swallow this. If a device was there before but we're dropping it now, seems like we should point give an indication of what we did. The same way that the other code path in the `else` clause does.

Reply:
I don't understand this: 
>If a device was there before but we're dropping it now

As I understand the code, when we first saw the device (in `OnNativeDeviceDiscovered`, we could not create a device for it, but did not throw (`OnNativeDeviceDiscovered` calls `AddDevice` with `throwIfNoLayoutFound: false`), and added it to `m_AvailableDevices`.

Now, on domain reload, we are processing the saved available devices again - and we _still_ cannot create a device for it. Only this time, we'd throw (`RestoreDevicesAfterDomainReload` calls AddDevice` with `throwIfNoLayoutFound: true`) - which did not make sense to me. So I changed it to not throw in this case either. It should still stay in `m_AvailableDevices`.

Also not sure what you mean by 
>The same way that the other code path in the `else` clause does.

what other code, what `else` clause?

Reply:
>when we first saw the device (in `OnNativeDeviceDiscovered`, we could not create a device for it, but did not throw (`OnNativeDeviceDiscovered` calls `AddDevice` with `throwIfNoLayoutFound: false`), and added it to `m_AvailableDevices`

`m_SavedDeviceStates` works differently. It actually *only* contains devices we successfully created before. It is `m_SavedAvailableDevices` which has the list of *all* devices that were reported.

So, when we fail here we are failing on something that we actually did *not* fail on before. Which is why I'm thinking it's good to give an indication of "hey user, we just dropped something that was there before".

For the HIDs in particular, these messages have actually successfully pointed out a bug. And even in cases where we rightfully dropped something because we no longer have a device layout matching the device, I found it useful to have feedback (not sure a user would say the same, though).

>what other code, what `else` clause?

Line 3275 and on. On that path, we check explicitly *before* trying to create a device and log a message if the layout is no longer there.

Reply:
Ok, got it. Will investigate further.